hal/api/platform.h

```diff
@@ -54,4 +55,52 @@ class PlatformMutex {
 
 #endif
 
+/** The static version of a PlatformMutex
+ *
+ * This class must only be used in a static context -
+ * this class must never be allocated or created on the
+ * stack.
+ *
+ * This class is lazily initialized on first use.
+ * This class is a POD type so if it is not used it will
+ * be garbage collected.
+ */
+struct PlatformMutexStatic {
+    PlatformMutex* _mutex;
+
+    void _init() {
+        PlatformMutex* temp = new PlatformMutex;
+
+        core_util_critical_section_enter();
+        if (NULL == _mutex) {
+            _mutex = temp;
+        }
+        core_util_critical_section_exit();
```

Comment:
Perhaps we should provide this functionality as a cas intrinsic?
``` c
core_util_critical_cas(unsigned *pointer, unsigned expected, unsigned desired);
```
prior art: http://en.cppreference.com/w/c/atomic/atomic_compare_exchange